### ISSUE TITLE
Config setting to reload dictionary on each call

### ISSUE THREAD

**Issue by elvarb** [logstash-plugins/logstash-filter-translate#42]
The use case I have for this is that I have a translate filter with a dictionary file and if no match is found I want to have a conditional that runs ruby code to fetch some data and update the dictionary file, then I want to have a different translate filter process the message with the updated dictionary file.

I could set that translate filter to reload every second but since this condition will not be hit very often it seems it would cause unnecessary load when I just want it to update when it runs, every time.

Is it possible to set the refresh interval to 0 for this ?

**Comment by xoh** [comment]
### TL;DR
> Is it possible to set the refresh interval to 0 for this ?

Yup.

### Long answer
As far as I can see, the dictionary is only reloaded, when the time passed since last reload is more than the refresh interval **and** an event is filtered, as the check if an update is required happens right before processing an event.

So, setting the refresh interval to 0 leads to refreshing the dictionary for _almost_ every event. Only two consecutive events, being processed within the very same nanosecond, could lead to skipping one refresh, but this should be virtually impossible.

I know this response comes very late, but it may help others with a similar use case. This issue can be closed, I guess?
